### DISCUSSION TITLE
$inertia.back()

### DISCUSSION THREAD

**Discussion by dillingham** [inertiajs/inertia#562] [Ideas]
In working with Laravel Jetstream I realized how much I loved having `back()` in vue-router

Very useful for `Cancel` buttons in forms that could be reached from multiple locations.

Linking to edit from index & show, cancel cant just link to `index` or `show` without disrupting the flow.

Would pair nicely with modals in https://github.com/inertiajs/inertia/discussions/584

**Comment by reinink** [comment]
Hey Brian, is there anything wrong with just using [History.back()](https://developer.mozilla.org/en-US/docs/Web/API/History/back)?

```js
window.history.back()
```

**Reply by dillingham** [reply]
Hey @reinink, no that works but you have to make a method:

```js
back() {
    window.history.back()
}
```

Whereas with $inertia.visit, its global so you can link things like
```html
<tr @click="$inertia.visit(route('posts.show', post.id))">
    <td>{{ post.title }}</td>
</tr>
```

Having `back` on `$inertia` would save having that method in each form component.

```html
<button @click="$inertia.back()">Cancel</button>
```

**Reply by dillingham** [reply]
Originally got the idea from Laravel Nova's source
```html
<cancel-button @click="$router.back()" />
```

**Reply by dillingham** [reply]
Another consideration: $inertia.back() within modals. 
Idk how a modal would be added into history

**Reply by claudiodekker** [reply]
Haha, yeah, Modals are a known (& admittedly fairly complex) 'todo' 👍 

For now, I'd recommend to just do something like this in your `app.js`:
```js
// Vue 2
Vue.prototype.window = window

// Vue 3
app.config.globalProperties.window = window

```

Which would allow you to do this in your templates:
```html
<span @click="window.history.back()">Go Back</span>
```

**Reply by dillingham** [reply]
@claudiodekker thanks :) That works. For my current purposes.. Im building a generator for jetstream and wanted to include a cancel button in the form it outputs.. and was trying to avoid adding a method or edit any other files. but thanks for the globalProperties.. didn' know that.
